moveit_ros/manipulation/pick_place/src/place.cpp

```diff
@@ -59,7 +59,9 @@ struct OrderPlaceLocationQuality
 
   bool operator()(const std::size_t a, const std::size_t b) const
   {
-    return places_[a].quality > places_[b].quality;
+    if (places_[a].quality > places_[b].quality) return true;
+    if (places_[a].quality == places_[b].quality) return a > b;
```

Comment:
Sorry, `sort` expects a less operator. The name `OrderPlaceLocationQuality` should reflect this...
What about the name `PlaceLocationLess`?
```suggestion
    if (places_[a].quality == places_[b].quality) return a < b;
```
You should definitely verify that this is the right combination. I just suggested this from the back of my head.

Reply:
What about `std::stable_sort` with the original compare operator?

Reply:
`stable_sort` [erzeugt nochmal einen temporären buffer](https://github.com/gcc-mirror/gcc/blob/d9375e490072d1aae73a93949aa158fcd2a27018/libstdc%2B%2B-v3/include/bits/stl_algo.h#L4977)... Daher ist die vorgeschlagene Lösung auf jeden Fall effizienter.

Reply:
German :smile: the temporary buffer should be negligible since the sorting algorithm sorts the vector of `std::size_t`, not the `PlaceLocations` itself. Plus using `stable_sort` should be easier to understand than a rather complicated compare logic.

Reply:
The cost of that buffer is surely negligible considering the rest of this pipeline. In favor of `stable_sort` (with a comment if necessary).

Also a quick reminder that this is slated to be replaced by MTC, so this can be revisited to your heart's content :0

Reply:
Ok, if there are two votes pro, go for it.

Reply:
Should I change `sort` to `stable_sort` in pick.cpp as well?

Reply:
Makes sense. I wouldn't expect the order to change...

Reply:
Done